projects/ng-components/lib/tooltip/tooltip.directive.ts

```diff
@@ -1,234 +1,177 @@
+import { TooltipContentComponent, TooltipTrigger, TooltipTheme } from "./tooltip-content/tooltip-content.component";
+import { Subject, fromEvent } from "rxjs";
+import { distinctUntilChanged, takeUntil } from "rxjs/operators";
 import { isEqual } from "lodash";
+import { getPlacementName, POSITION_MAP, DEFAULT_TOOLTIP_POSITIONS, TooltipPosition, CASCADE_TOOLTIP_POSITIONS } from "./tooltip.positions";
 
-type Placement = {
-    [K in TooltipPosition]: ConnectedPosition;
-};
 /**
  * A text label that acts as a helper to a specific item
  */
 @Directive({ selector: "[sebng-tooltip]" })
-export class TooltipDirective implements OnInit, OnDestroy {
+export class TooltipDirective implements OnDestroy {
     /** content of tooltip */
     @Input("sebng-tooltip") content: string | TemplateRef<any> = "";
+
     /** tooltip trigger method */
     @Input() trigger: TooltipTrigger = "hover";
+
     /** tooltip position */
     @Input() position: TooltipPosition = "top";
     /** tooltip theme */
     @Input() theme: TooltipTheme = "default";
     /** CSS class */
     @Input() className?: string = "";
+    /** reposition tooltip with more placement choices */
+    @Input() cascade?: boolean;
+
+    /** Close the tooltip once the user scrolls */
+    @Input() closeOnScroll: boolean = false;
+
+    /** Delay before closing on scroll */
+    @Input() closeOnScrollDelay: number = 0;
 
-    @HostBinding("attr.tabindex") tabindex = -1;
     /** <!-- skip --> */
     private overlayRef: OverlayRef;
     /** <!-- skip --> */
     private tooltipRef: ComponentRef<TooltipContentComponent>;
+
     /** <!-- skip --> */
-    private placements: Placement = {
-        left: this.getOriginPosition("start", "center", "end", "center"),
-        "left-top": this.getOriginPosition("start", "top", "end", "top"),
-        "left-bottom": this.getOriginPosition("start", "bottom", "end", "bottom"),
-        bottom: this.getOriginPosition("center", "bottom", "center", "top"),
-        "bottom-left": this.getOriginPosition("start", "bottom", "start", "top"),
-        "bottom-right": this.getOriginPosition("end", "bottom", "end", "top"),
-        right: this.getOriginPosition("end", "center", "start", "center"),
-        "right-top": this.getOriginPosition("end", "top", "start", "top"),
-        "right-bottom": this.getOriginPosition("end", "bottom", "start", "bottom"),
-        "top-left": this.getOriginPosition("start", "top", "start", "bottom"),
-        "top-right": this.getOriginPosition("end", "top", "end", "bottom"),
-        top: this.getOriginPosition("center", "top", "center", "bottom"),
-    };
-    /** <!-- skip --> */
-    private obs: Subscription;
+    destroy$: Subject<boolean> = new Subject<boolean>();
 
-    constructor(private overlay: Overlay, private overlayPositionBuilder: OverlayPositionBuilder, private elementRef: ElementRef) {}
+    constructor(private overlay: Overlay, private elementRef: ElementRef, private renderer: Renderer2) {}
```

Comment:
It seems that you are not using the `renderer` anywhere in the code. If that's the case, maybe you can remove it to reduce the bundle size.

Reply:
yea, removed it